File: Vigilantes.DaprWorkshop.LoyaltyService/node/controllers/LoyaltyController.js
const LoyaltySummary = require('../models/LoyaltySummary');

// POST /orders
exports.orders = async function(req, res) {
  const orderSummary = req.body;
  console.log(`OrderId = ${orderSummary.orderId}, StoreId = ${orderSummary.storeId}, CustomerName = ${orderSummary.customerName}, LoyaltyId = ${orderSummary.loyaltyId}, OrderItemCount = ${orderSummary.orderItems.length}, OrderTotal = ${orderSummary.orderTotal}`);
  
  // TODO: Challenge 3 - Retrieve and update customer loyalty points
  res.json();
};

File: Vigilantes.DaprWorkshop.LoyaltyService/csharp/Controllers/LoyaltyController.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using CloudNative.CloudEvents;
using Dapr;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json.Linq;
using Vigilantes.DaprWorkshop.LoyaltyService.Models;

namespace Vigilantes.DaprWorkshop.LoyaltyService.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class LoyaltyController : ControllerBase
    {
        private readonly HttpClient _httpClient;
        private readonly ILogger<LoyaltyController> _logger;

        public LoyaltyController(IHttpClientFactory httpClientFactory, ILogger<LoyaltyController> logger)
        {
            _httpClient = httpClientFactory.CreateClient();
            _logger = logger;
        }

        [HttpPost("/orders")]
        [Topic("pubsub", "newOrder")]
        public async Task<IActionResult> UpdateLoyalty(CloudEvent cloudEvent)
        {
            var orderSummary = ((JToken)cloudEvent.Data).ToObject<OrderSummary>();
            _logger.LogInformation("Received Order Summary: {@OrderSummary}", orderSummary);

            // TODO: Challenge 3 - Retrieve and update customer loyalty points
            return Ok();
        }
    }
}
